chcemvediet/apps/anonymization/cron.py

```diff
@@ -25,6 +33,47 @@ def attachment_normalization():
             content_type=content_types.PDF_CONTENT_TYPE
         )
         cron_logger.info(u'Normalized attachment: {}'.format(attachment))
+
+    elif attachment.content_type in content_types.LIBREOFFICE_CONTENT_TYPES:
+        directory_name = tempfile.mkdtemp(dir=u'.')
+        file_copy = os.path.join(directory_name, u'file' + guess_extension(attachment.content_type))
+        shutil.copy2(attachment.file.path, file_copy)
+
+        p = subprocess32.Popen(
+            [u'libreoffice', u'--headless', u'--convert-to', u'pdf', u'--outdir', directory_name,
+             file_copy],
+            stdout=subprocess32.PIPE,
+            stderr=subprocess32.PIPE
+        )
+        try:
+            out, err = p.communicate(timeout=300)
+        except subprocess32.TimeoutExpired:
+            p.terminate()
+            out, err = p.communicate()
+            err += u'Timeout expired.'
+
+        if u'file.pdf' in os.listdir(directory_name):
+            file_pdf = open(os.path.join(directory_name, u'file.pdf'), u'rb')
```

Comment:
Opat `open` zabal do context managera, aby sa subor na konci automaticky zatvoril. Vsetko, co otvara/vytvara nejake externe resourcy, ktore treba na konci upratat, by sa malo robit cez context managery. Nech je istota, ze sa to uprace aj ked v kode nieco failne.

Reply:
:ok: